pmd-jsp/src/main/ant/alljavacc.xml

```diff
@@ -120,17 +120,17 @@ public class Token implements GenericToken, java.io.Serializable]]></replacevalu
 
 
         <replaceregexp>
-            <regexp pattern="class|interface" />
-            <substitution expression="@Deprecated @net.sourceforge.pmd.annotation.InternalApi \0" />
+            <regexp pattern="public (?:class|interface)" />
+            <substitution expression="@Deprecated${line.separator}@net.sourceforge.pmd.annotation.InternalApi${line.separator}\0" />
```

Comment:
```suggestion
            <substitution expression="@Deprecated\R@net.sourceforge.pmd.annotation.InternalApi\R\0" />
```

This would match both \r and \r\n regardless of the host system, which might be slightly robuster?

Reply:
I've tried it - it doesn't work. \R seems only be supported in the pattern, but not in the substitution....

Reply:
Good to know